Sources/GoogleAI/FunctionCalling.swift

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// A predicted function call returned from the model.
+///
+/// REST Docs: https://ai.google.dev/api/rest/v1beta/Content#functioncall
+public struct FunctionCall: Equatable, Encodable {
+  /// The name of the function to call.
+  public let name: String
+
+  /// The function parameters and values.
+  public let args: JSONObject
+}
+
+// REST Docs: https://ai.google.dev/api/rest/v1beta/Tool#schema
+public class Schema: Encodable {
```

Comment:
Can this be "FunctionCallSchema"?  "Schema" by itself doesn't mean a lot to me.

Reply:
The naming of all of these objects can definitely be changed. I was just holding off on naming until we settled on the design.

Reply:
I retract this. Anyone using this will need to read the Google AI docs and matching makes sense.